l10n_ch_lsv_dd/wizards/lsv_export_wizard.py

```diff
@@ -0,0 +1,485 @@
+        'Number of Transactions',
+        related='banking_export_ch_dd_id.nb_transactions'
+    )
+    total_amount = fields.Float(
+        related='banking_export_ch_dd_id.total_amount'
+    )
+    state = fields.Selection(
+        [('create', _('Create')), ('finish', _('Finish'))],
+        readonly=True,
+        default='create'
+    )
+
+    @api.multi
+    def generate_lsv_file(self):
+        ''' Generate direct debit export object including the lsv file
+            content. Called by generate button.
+        '''
+        self.ensure_one()
+        payment_order_obj = self.env['account.payment.order']
+        payment_line_obj = self.env['bank.payment.line']
+
+        active_ids = self.env.context.get('active_ids', [])
+        if not active_ids:
+            raise exceptions.ValidationError(_('No payment order selected'))
+        payment_orders = payment_order_obj.browse(active_ids)
+
+        # common properties for all lines
+        properties = self._setup_properties(payment_orders[0])
+
+        total_amount = 0.0
+        lsv_lines = []
+
+        for payment_order in payment_orders:
+            total_amount = total_amount + payment_order.total_company_currency
+
+            ben_bank = payment_order.company_partner_bank_id
+            clean_acc_number = ben_bank.acc_number.replace(' ', '')
+            clean_acc_number = clean_acc_number.replace('-', '')
+            ben_address = self._get_account_address(ben_bank)
+            properties.update({
+                'ben_address': ben_address,
+                'ben_iban': clean_acc_number,
+                'ben_clearing':
+                    self._get_clearing(payment_order.company_partner_bank_id),
+            })
+
+            if not self._is_ch_li_iban(properties.get('ben_iban')):
+                raise exceptions.ValidationError(
+                    _('Ben IBAN is not a correct CH or LI IBAN (%s given)') %
+                    properties.get('ben_iban')
+                )
+
+            order_by = ''
+            if payment_order.date_prefered == 'due':
+                order_by = 'm.date_maturity ASC, '
+            order_by += 'al.partner_bank_id'
+
+            # A direct db query is used because order parameter in model.search
+            # doesn't support function fields
+            self.env.cr.execute("""
+                SELECT bl.id, m.date_maturity, al.partner_bank_id
+                FROM bank_payment_line bl
+                JOIN account_payment_line al ON al.bank_line_id = bl.id
+                JOIN account_move_line m ON al.move_line_id = m.id
+                WHERE bl.order_id = %s
+                ORDER BY """ + order_by, (payment_order.id,))
+            sorted_line_ids = [row[0] for row in self.env.cr.fetchall()]
+            payment_lines = payment_line_obj.browse(sorted_line_ids).exists()
+
+            for line in payment_lines:
+                if not line.mandate_id or not line.mandate_id.state == "valid":
```

Comment:
`not ==` can be replaced by `!=`

And you can remove the test `line.mandate_id`

line will always be a recordset, and when you ask for a relation of a recordset, (even empty) you will get a record set.


```
>>> bool(line.mandate_id) == False
True
>>> line.mandate_id
account.banking.mandate()
>>> line.mandate_id.state
False
```

So getting an empty recordset will allow you to have your tests on the attributes of the relation without fearing it will crash.